controllers/limitador_controller.go

```diff
@@ -278,3 +290,22 @@ func mutateLimitadorDeployment(existingObj, desiredObj client.Object) (bool, err
 
 	return updated, nil
 }
+
+func getStorageConfigSecret(ctx context.Context, client client.Client, secretRef *v1.ObjectReference) (*v1.Secret, error) {
+	storageConfigSecret := &v1.Secret{}
+	if err := client.Get(
+		ctx,
+		types.NamespacedName{
+			Name:      secretRef.Name,
+			Namespace: secretRef.Namespace,
+		},
+		storageConfigSecret,
+	); err != nil {
+		return nil, err
+	}
+
+	if len(storageConfigSecret.Data) > 0 && storageConfigSecret.Data["URL"] != nil {
+		return storageConfigSecret, nil
+	}
+	return nil, fmt.Errorf("the storage config Secret doesn't have the `URL` field")
```

Comment:
if there is not formatting, `errors.New` works

Reply:
What would be the equivalent in  `k8s.io/apimachinery/pkg/api/errors` module for returning an `error` like Go core module `New` method ?

Reply:
no need to be `k8s.io/apimachinery/pkg/api/errors`. It can be go builtin error. They all implement Error interface

Reply:
OK, I could include the go builtin error lib